File: frontend/vitruvius-editor/src/model/DisplayView.ts
/**
 * Represents a display view with various selector and mapper names.
 */
export type DisplayView = {
  /**
   * The name of the display view.
   */
  name: string;

  /**
   * The name of the view type.
   */
  viewTypeName: string;

  /**
   * The name of the view mapper.
   */
  viewMapperName: string;

  /**
   * The name of the window selector.
   */
  windowSelectorName: string;

  /**
   * The name of the content selector.
   */
  contentSelectorName: string;
};

File: backend/src/main/kotlin/tools/vitruv/vitruvAdapter/core/api/DisplayView.kt
package tools.vitruv.vitruvAdapter.core.api

/**
 * This interface represents a display view, which can be displayed in the Vitruvius graphical editor.
 * @author uhsab
 */
interface DisplayView {
    /**
     * The name of the display view.
     */
    val name: String
    /**
     * The name of the viewType for the displayView.
     */
    val viewTypeName: String
    /**
     * The view mapper for the display view.
     */
    val viewMapper: ViewMapper<Any?>
    /**
     * The internal selector for the display view.
     * This selector is used to filter elements needed for the display view.
     */
    val internalSelector: Selector
    /**
     * The content selector for the display view.
     * This selector is used to select the content for the display view, for specific windows.
     */
    val contentSelector: ContentSelector
}


